File: 121_best_time_to_buy_and_sell_stock/scala/src/test/scala/TestSolution.scala
/*
 * leetcode
 * 121. Best Time to Buy and Sell Stock
 * Tests
 */

import org.scalatest.flatspec.AnyFlatSpec
import Solution.maxProfit

class TestSolution extends AnyFlatSpec {
  behavior of "Solution.maxProfit"

  it should "find max profit to be zero in special cases" in {
    // zero prices
    assert(maxProfit(Array()) == 0)
    // only 1 price
    assert(maxProfit(Array(99)) == 0)
    // all prices are the same
    assert(maxProfit(Array.fill(10)(22)) == 0)
  }

  it should "find max profit buying and selling stock" in {
    assert(maxProfit(Array(9, 5, 8, 6, 2, 1, 4)) == 3)
    assert(maxProfit(Array(9, 8, 7, 6, 5, 2, 1)) == 0)
    assert(maxProfit(Array(3, 4, 8, 9, 10)) == 7)
    assert(maxProfit(Array(1, 0, 10, 2, 50, 99, 21)) == 99)
  }
}

File: 217_contains_duplicate/test_solution.py
#
# Leetcode
# 217. Contains Duplicate
# Tests
#

from solution import Solution
from pytest import fixture


@fixture
def solution():
    return Solution()


def test_containsDuplicate_special_cases(solution: Solution):
    # test special cases
    assert solution.containsDuplicate([]) == False
    assert solution.containsDuplicate([1]) == False
    assert solution.containsDuplicate([1, 1, 1]) == True


def test_containsDuplicate(solution: Solution):
    assert solution.containsDuplicate([1, 2, 3, 4, 4]) == True
    assert solution.containsDuplicate([5, 2, 8, 5, 2]) == True
    assert solution.containsDuplicate([1, 8, 8, 5, 9]) == True
    assert solution.containsDuplicate([1, 2, 3, 9, 5]) == False
